lib/reports/subscriber_lists_report.rb

```diff
@@ -2,7 +2,12 @@ class Reports::SubscriberListsReport
   attr_reader :date, :slugs, :tags_pattern, :links_pattern, :headers
 
   def initialize(date, slugs: "", tags_pattern: nil, links_pattern: nil, headers: nil)
-    @date = Time.zone.parse(date)
+    begin
+      @date = Date.parse(date)
+    rescue Date::Error
+      @date = nil
+    end
```

Comment:
Ah sorry I hadn't understood the delicate balance this has. 

It seems a touch convoluted to me to deal with a nil here that only raises an exception later. So this could be:

```
    begin
      @date = Date.parse(date)
    rescue Date::Error
      raise "invalid date" 
    end
```

but I don't really think that's necessary as it doesn't provide any more or less information than the `Date::Error` exception of `Date::Error (invalid date)`. 

So I'd suggest doing just `Date.parse(date)` here and removing https://github.com/alphagov/email-alert-api/blob/913a48b2816248182b2ae357a2644ea58b0cc487/lib/reports/subscriber_lists_report.rb#L39 and the associated test.